pkg/transport/kafka_sarama/receiver.go

```diff
@@ -17,22 +19,16 @@ type msgErr struct {
 // Receiver which implements sarama.ConsumerGroupHandler
 // After the first invocation of Receiver.Receive(), the sarama.ConsumerGroup is created and started.
 type Receiver struct {
+	once     sync.Once
 	incoming chan msgErr
-
-	client              sarama.Client
-	topic               string
-	groupId             string
-	saramaConsumerGroup sarama.ConsumerGroup
 }
 
 // NewReceiver creates a Receiver which implements sarama.ConsumerGroupHandler
+// The sarama.ConsumerGroup must be started invoking. If you need a Receiver which also manage the ConsumerGroup, use NewConsumer
 // After the first invocation of Receiver.Receive(), the sarama.ConsumerGroup is created and started.
-func NewReceiver(client sarama.Client, groupId string, topic string) *Receiver {
+func NewReceiver() *Receiver {
 	return &Receiver{
```

Comment:
do you need to export this?

Reply:
yes, in case someone wants to use the receiver without using the ConsumerGroup created by Consumer